src/services/schemas/SchemasService.ts

```diff
@@ -67,6 +67,7 @@ export class SchemaService {
         error => deferred.reject(error))
       .then(error => {
           if (isNotNullOrUndefined(error)) {
+            this.unregisterProtoSchema(container, schema.fileName);
```

Comment:
We should take it only if we get an error @ryanemerson See reopened ISPN-7228 for details